src/contexts/LocalStorage.js

```diff
@@ -116,20 +116,25 @@ export function useSavedAccounts() {
   const [state, { updateKey }] = useLocalStorageContext()
   const savedAccounts = state?.[SAVED_ACCOUNTS]
 
-  function addAccount(account) {
-    let newAccounts = state?.[SAVED_ACCOUNTS]
-    newAccounts.push(account)
-    updateKey(SAVED_ACCOUNTS, newAccounts)
-  }
+  const addAccount = useCallback(
+    (account) => {
+      updateKey(SAVED_ACCOUNTS, [...(savedAccounts ?? []), account])
+    },
+    [savedAccounts, updateKey]
+  )
 
-  function removeAccount(account) {
-    let newAccounts = state?.[SAVED_ACCOUNTS]
-    let index = newAccounts.indexOf(account)
-    if (index > -1) {
-      newAccounts.splice(index, 1)
-    }
-    updateKey(SAVED_ACCOUNTS, newAccounts)
-  }
+  const removeAccount = useCallback(
+    (account) => {
+      let index = savedAccounts?.indexOf(account) ?? -1
+      if (index > -1) {
+        updateKey(SAVED_ACCOUNTS, [
+          ...savedAccounts.slice(0, index),
+          ...savedAccounts.slice(index + 1, savedAccounts.length),
+        ])
+      }
+    },
+    [savedAccounts, updateKey]
+  )
```

Comment:
This change ensures that we do not mutate the original `savedAccounts` array and also creates stable references for the `addAccount` and `removeAccount` functions.

Reply:
Looks good to me - thanks for the contribution here (sorry for the late reply). This will be a good improvement.